pkg/operator/ceph/object/objectstore.go

```diff
@@ -758,26 +761,85 @@ func enableRGWDashboard(context *Context) error {
 	if err != nil || errCode != 0 {
 		return errors.Wrapf(err, "failed to create user %q", DashboardUser)
 	}
-	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
-	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
-	_, err = cephCmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to set user %q accesskey", DashboardUser)
-	}
 
-	args = []string{"dashboard", "set-rgw-api-secret-key", *u.SecretKey}
-	cephCmd = ceph.NewCephCommand(context.Context, context.clusterInfo, args)
-	go func() {
-		// Setting the dashboard api secret started hanging in some clusters
-		// starting in ceph v15.2.8. We run it in a goroutine until the fix
-		// is found. We expect the ceph command to timeout so at least the goroutine exits.
-		logger.Info("setting the dashboard api secret key")
-		_, err = cephCmd.RunWithTimeout(applyDashboardKeyTimeout)
+	cephVersion := mgr.CephVersionsNewCephDashboardCommand(context.clusterInfo)
+	// for latest Ceph versions
+	if cephVersion {
+		// Generate a temp file
+		file, err := ioutil.TempFile("", "")
```

Comment:
How about a helper method for lines 769-776 for creating the temp password file that is needed in several places?

Reply:
Done.